### PR TITLE
smash: use varnish for caching

### PR DESCRIPTION
**Pull request by jbgi**
need to merge https://github.com/input-output-hk/ops-lib/pull/51 first

### CODE REVIEW

File: nix/sources.json [hunk 1]
```diff
@@ -145,15 +145,15 @@
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
     },
     "ops-lib": {
-        "branch": "master",
+        "branch": "varnish",
```
Comment (johnalotoski) [inline nix/sources.json:148]: Can change this back to HEAD of master if https://github.com/input-output-hk/ops-lib/pull/51 is merged before this one.

File: roles/smash.nix [hunk 2]
```diff
@@ -94,25 +94,91 @@ in {
...
     recommendedOptimisation = true;
     recommendedProxySettings = true;
-    preStart = ''
-      [ -d "${nginxCachePath}" ] || { echo "The nginx data cache dir does not exist"; exit 1; }
-      [ -w "${nginxCachePath}" ] || { echo "The nginx data cache dir is not writable by nginx"; exit 1; }
-    '';
+    #preStart = ''
+    #  [ -d "${nginxCachePath}" ] || { echo "The nginx data cache dir does not exist"; exit 1; }
+    #  [ -w "${nginxCachePath}" ] || { echo "The nginx data cache dir is not writable by nginx"; exit 1; }
+    #'';
```
Comment (johnalotoski) [inline roles/smash.nix:167]: We can just remove this too if the rest of nginx caching is being deleted in favor of varnish.

File: roles/smash.nix [hunk 3]
```diff
@@ -94,25 +94,91 @@ in {
...
+        .host = "127.0.0.1";
+        .port = "3100";
+      }
+      acl purge {
+        "localhost";
+        "127.0.0.1";
+      }
+      sub vcl_recv {
+        unset req.http.x-cache;
+        # allow PURGE from localhost and 192.168.55...
```
Comment (johnalotoski) [inline roles/smash.nix:114]: We can drop the `and 192.168.55...` as it's no longer included in this code.

File: roles/smash.nix [hunk 4]
```diff
@@ -94,25 +94,91 @@ in {
...
+    #preStart = ''
+    #  [ -d "${nginxCachePath}" ] || { echo "The nginx data cache dir does not exist"; exit 1; }
+    #  [ -w "${nginxCachePath}" ] || { echo "The nginx data cache dir is not writable by nginx"; exit 1; }
+    #'';
     commonHttpConfig = let
       apiKeys = import ../static/smash-keys.nix;
       allowedOrigins = lib.optionals (builtins.pathExists ../static/smash-allow-origins.nix) (import ../static/smash-allow-origins.nix);
     in ''
-      log_format x-fwd '$remote_addr - $remote_user $upstream_cache_status [$time_local] '
+      log_format x-fwd '$remote_addr - $remote_user [$time_local] '
```
Comment (johnalotoski) [inline roles/smash.nix:172]: Hit will be excluded from the logs via $loggable, but won't we still want to log the caching status here from varnish (miss, pass, pipe, purge) for insight into what isn't being hit and why?

File: roles/smash.nix [hunk 5]
```diff
@@ -94,25 +94,91 @@ in {
...
                        '"$request" $status $body_bytes_sent '
                        '"$http_referer" "$http_user_agent" "$http_x_forwarded_for"';
 
-      access_log syslog:server=unix:/dev/log x-fwd if=$not_cached;
+      map $upstream_cache_status $loggable {
+          "HIT" 0;
```
Comment (johnalotoski) [inline roles/smash.nix:177]: Is this working properly? The `HIT` text here is uppercase while in varnish config, it's lowercase `hit`.

Also is use of `$upstream_cache_status` correct?  Headers from an upstream caching server can also be obtained with `$upstream_http_*name*` it looks like.

Comment (jbgi) [reply roles/smash.nix:177]: yeah, I think this config is not working, at least it does not make any difference. 
I'll remove it for now, and see if logs need more cleanup.

File: roles/smash.nix (hunk 4 above)
Comment (jbgi) [reply roles/smash.nix:172]: cc @disassembler

Comment (disassembler) [reply roles/smash.nix]: I think the $loggable stuff got dropped along the way... Added it back in and ignore "hit cached" of x-cache header from logs, as well as in the process figured out how to remove the massive metric spam from logs as well.

**Review by disassembler — APPROVED** [review]

**Pull request merged**